packages/flutter_tools/test/src/testbed.dart

```diff
@@ -135,11 +160,257 @@ class NoOpUsage implements Usage {
   void sendCommand(String command, {Map<String, String> parameters}) {}
 
   @override
-  void sendEvent(String category, String parameter, {Map<String, String> parameters}) {}
+  void sendEvent(String category, String parameter,{ Map<String, String> parameters }) {}
 
   @override
   void sendException(dynamic exception, StackTrace trace) {}
 
   @override
-  void sendTiming(String category, String variableName, Duration duration, {String label}) {}
-}
\ No newline at end of file
+  void sendTiming(String category, String variableName, Duration duration, { String label }) {}
+}
+
+class FakeHttpClient implements HttpClient {
```

Comment:
Can we remove or DRY out the other places attempting this since it's already mocked in the outer zone? MockHttpClient, MockHttpClientRequest, MockHttpClientResponse in create_test.dart and context.dart, _MockHttpClient in binding, etc.

And

Reply:
The Mocks and Fakes shouldn't be used interchangeably. The former have the potential to have behaviors added dynamically (via mock/noSuchMethod) that can leak between tests, while the later is safe to reuse across test runs.

The _MockHttpClient in binding.dart should be renamed _FakeHttpClient (that one is my fault). But it can't be shared because flutter_test has a dependency on dart:ui, which doesn't exist on the command line where the tool runs.

Any cleanup we do would be to simplify/remove the mocks in context and create_test.

Reply:
Bleh it's a shame fakes is How It's Done with Dart testing since now you're on the hook for overriding all new methods, and also a shame that [MockClient](https://pub.dev/documentation/http/latest/testing/MockClient-class.html) doesn't look very useful.

Reply:
Ideally the HttpClient* classes would be extendable, but they are just interfaces so we're on the hook for everything.